src/open_company_web/lib/utils.cljs

```diff
@@ -504,20 +504,37 @@
     (> current-month 9)
     10))
 
+(defn non-zero-number?
+  [number]
+  ; Seems a little convoluted, but need it to short circuit if it's not a number
+  (not (or (not (number? number)) (zero? number))))
+
+(defn no-finances-data?
+  "Return `true` if the passed finances data has no data, otherwise return `false`."
+  [data]
+  (not (some #(or (non-zero-number? (:cash %))
+                  (non-zero-number? (:revenue %))
+                  (non-zero-number? (:costs %))) data)))
+
+(defn no-growth-data?
+  "Return `true` if the passed finances data has no data, otherwise return `false`."
+  [data]
+  (not (some #(non-zero-number? (:value %)) (vals data))))
+
 (defn current-growth-period [interval]
   (let [now (cljs-time/now)
         year (cljs-time/year now)
         month (cljs-time/month now)]
     (case interval
       "quarterly"
       (str year "-" (add-zero (get-month-quarter month)))
-      "monthly"
-      (str year "-" (add-zero month))
       "weekly"
       (let [day-of-week (cljs-time/day-of-week now)
             to-monday (dec day-of-week)
             monday-date (cljs-time/minus now (cljs-time/days to-monday))]
-        (str (cljs-time/year monday-date) "-" (add-zero (cljs-time/month monday-date)) "-" (add-zero (cljs-time/day monday-date)))))))
+        (str (cljs-time/year monday-date) "-" (add-zero (cljs-time/month monday-date)) "-" (add-zero (cljs-time/day monday-date))))
+      ;; Default tp monthly
+      (str year "-" (add-zero month)))))
```

Comment:
Not a fan of formatting things by hand like this when we have things like `goog.date.Date` and `goog.i18n.DateTimeFormat`. Even cljs-time which is used here has formatting capabilities: https://github.com/andrewmcveigh/cljs-time#cljs-timeformat